scrapy/utils/datatypes.py

```diff
@@ -69,16 +70,35 @@ class LocalCache(collections.OrderedDict):
     Older items expires first.
     """
 
-    def __init__(self, limit=None):
+    def __init__(self, limit=None, time_limit=None):
         super().__init__()
         self.limit = limit
+        self.time_limit = time_limit
 
     def __setitem__(self, key, value):
-        if self.limit:
+        now = datetime.now()
+        value = (value, now)
+
+        if not self.limit is None:
```

Comment:
if self.limit is not None. Please refer to: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
I didn't know about this, thanks!
I changed all `not ... is None` to `... is not None`